### PR TITLE
Create bashible failure events in the default namespace

### PR DESCRIPTION
**Pull request by Diaderius**
## Description
Create bashible failure events in the default namespace.

## Why do we need it, and what problem does it solve?
It is feature regarding for creation events in the **default** namespace if cluster isn't bootstrapped. By this way, you doesn't need to connect to node directly - the event with reference to bashible failure step should be created.
It refers to: #2827

## What is the expected result?
If bashible step is failed, it calls the creating event with bashible step failure in the **default** namespace.

## Checklist
- [ ] The code is covered by unit tests.
- [x] e2e tests passed.
- [ ] Documentation updated according to the changes.
- [x] Changes were tested in the Kubernetes cluster manually.

## Changelog entries

```changes
section: candi
type: feature
summary: create bashible events with errors in the default namespace
```

### CODE REVIEW

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @Diaderius. I've started the workflow for label 'e2e/run/azure'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @Diaderius. I've started the workflow for label 'e2e/run/aws'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @Diaderius. I've started the workflow for label 'e2e/run/azure'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @Diaderius. I've started the workflow for label 'e2e/run/aws'...

**Review by RomanenkoDenys — CHANGES_REQUESTED** [review]
need some changes as discussed

File: candi/bashible/bashbooster/20_event.sh [hunk 1]
```diff
@@ -122,3 +122,23 @@ bb-event-cleanup() {
     fi
     BB_EVENT_DEPTH["__delay__"]=$(( ${BB_EVENT_DEPTH["__delay__"]} - 1 ))
 }
+bb-event-error-create() {
+    eventHash="$(echo -n "$(hostname -s)")-$(journalctl -xeu bashible.service -n 7 | grep -e "Step" | sed -r 's/.*\/(.*)\.sh/\1/' | md5sum | awk '{print $1}' | cut -b -5)"
```
Comment (yalosev) [inline candi/bashible/bashbooster/20_event.sh:126]: I think we need a comment here to describe - what are we doing here?

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @Diaderius. I've started the workflow for label 'e2e/run/aws'...

File: candi/bashible/bashible.sh.tpl [hunk 2]
```diff
@@ -20,6 +20,30 @@ function kubectl_exec() {
...
+        metadata:
+          name: bashible-error-${eventName}
+          namespace: d8-cloud-instance-manager
+        reason: Failed
+        type: Error
+        lastTimestamp: '$(date -u +"%Y-%m-%dT%H:%M:%SZ")'
+        message: '$(cat "${eventLog}" | tail -c 500)'
+        involvedObject:
+          kind: Node
+          namespace: d8-cloud-instance-manager
```
Comment (yalosev) [inline candi/bashible/bashible.sh.tpl:39]: Do we have a namespace for Node object?

Comment (Diaderius) [reply candi/bashible/bashible.sh.tpl]: fixed to default ns.

File: candi/bashible/bashible.sh.tpl [hunk 3]
```diff
@@ -20,6 +20,30 @@ function kubectl_exec() {
   kubectl --request-timeout 60s --kubeconfig=/etc/kubernetes/kubelet.conf ${@}
 }
 
+function bb-event-error-create() {
+    # eventName parameter aggregates hostname with bashible step name
+    eventName="$(echo -n "$(hostname -s)")-${step}"
+    eventLog="/var/lib/bashible/step.log"
```
Comment (RomanenkoDenys) [inline candi/bashible/bashible.sh.tpl:26]: step should be an argument of the bb-event-error-create function

**Review by RomanenkoDenys — COMMENTED** [review]
one simple change

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @Diaderius. I've started the workflow for label 'e2e/run/aws'...

**Review by RomanenkoDenys — APPROVED** [review]
LGTM

**Review by yalosev — APPROVED** [review]

**Pull request merged**